### PR TITLE
Prepare v1.1.4

### PR DESCRIPTION
**Pull request by pasanw**
Picks the CVE fix commits that occurred in master after v1.1.3 was cut. All commits after v1.1.3 can be viewed [here](https://github.com/tigera/key-cert-provisioner/compare/0aea3d7b05a963385ca0424e681c34c5010072fd...master).

Specifically, this picks all the commits, except:
* The FIPS commit (therefore, any UBI-related changes are no longer applied)
* CodeQL configuration commits
* Merge commits (verified to be pure merge commits)

I performed a scan comparing the current hashrelease image (v1.2.0) against the new proposed image [here](https://tigera.semaphoreci.com/workflows/d1f60e2a-e76a-4b2a-941b-ac985f38f0d8?pipeline_id=9a389a80-188b-4100-978b-5f0c503957cc)

### CODE REVIEW

**Comment by Behnam-Shobiri** [comment]
LGTM, Thanks for cherry-picking the fixes @pasanw

**Review by Behnam-Shobiri — APPROVED** [review]

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and leave the branch after I've merged it.

**Review by matthewdupre — APPROVED** [review]

**Pull request merged**
